flashmodels/patch/patch.py

```diff
@@ -41,18 +41,17 @@ def patch_llama():
         transformers.models.llama.modeling_llama.LlamaAttention.forward = flash_attn_fwd
     elif os.environ.get("ACC_LLAMA_TP") == "1":
         transformers.models.llama.modeling_llama.LlamaMLP = LlamaMLP
-    if os.getenv("XLA_USE_SPMD") == "1":
-        # use einsum in linear for SPMD TP/Ulysses.
-        transformers.models.llama.modeling_llama.LlamaAttention = LlamaAttention
-        transformers.models.llama.modeling_llama.LlamaDecoderLayer = LlamaDecoderLayer
+    # if os.getenv("XLA_USE_SPMD") == "1":
```

Comment:
do we need export XLA_USE_SPMD=1 when using spmd_fsdp?

Reply:
replace with `use_tp` args.